File: Python/8py 1/dna_to_rna.py
"""
|--------------------------------------------------------------------------
| DNA to RNA Conversion
|--------------------------------------------------------------------------
|
"""

def dna_to_rna(dna):
   return 'U'.join(dna.split('T'))


def DNAtoRNA(dna):
   return dna.replace('T', 'U')


import re
def DNAtoRNA(dna):
   return re.sub('T', 'U', dna)


def DNAtoRNA(dna):
   return dna.translate(dna.maketrans("T", "U"))


print(dna_to_rna("TTTT"), "UUUU")
print(dna_to_rna("GCAT"), "GCAU")
print(dna_to_rna("GACCGCCGCC"), "GACCGCCGCC")

File: JavaScript/8js 1/DNAtoRNA.js
/*
|--------------------------------------------------------------------------
| DNA to RNA Conversion
|--------------------------------------------------------------------------
|
*/

function DNAtoRNA(dna) {
   return dna.replace(/T/g, 'U')
}


const DNAtoRNA = dna => dna.split('').map(i => i == 'T' ? i = 'U' : i).join('');


function DNAtoRNA(dna) {
   return dna.split("T").join("U");
}


function DNAtoRNA(dna) {
   let result = '';
   for (let i = 0; i < dna.length; i++) {
      dna[i] == 'T' ? result += 'U' : result += dna[i];
   }
   return result;
}


console.log(DNAtoRNA("TTTT"), "UUUU")
console.log(DNAtoRNA("GCAT"), "GCAU")
console.log(DNAtoRNA("GACCGCCGCC"), "GACCGCCGCC")